### ISSUE TITLE
[Feature request] Add mobile keyboard input support

### ISSUE THREAD

**Issue by yujinqiu** [m1k1o/neko#115]
neko's mobile support is good,  but  when I  try to input URL in neko chrome (m1k1o/neko:google-chrome) not keyboard show up.

**Comment by m1k1o** [comment]
Mobile support is just partial for now, that means at least watching is possible. That was not the case before, so we are going slowly in that direction, but there are certain things missing:
- Moving cursor around the screen. The cursor would be fixed and when swapping to the side it would move to the side. Right now only the touch position is sent, that would need to be changed.
- Keyboard cannot show up because n.eko does not know when a keyboard is needed. We would need to get that event from Xorg and forward it to client, that text input is active.
- The keyboard events are sent in uint64 which is not supported by Safari, in order to get it working there we need to change to uint32. I checked, the data field is oversized now, it uses only 29 bits, so uint32 would be sufficient. https://github.com/m1k1o/neko/blob/23ad1cf882ba9f639a7af97747a34812707e415b/client/src/neko/base.ts#L155-L170

**Comment by akashsharma95** [comment]
@m1k1o can I take this up?

**Comment by m1k1o** [comment]
@akashsharma95 yes, sure.

**Comment by dpalmigiano** [comment]
So why not enable the keyboard by default? The key events could go to the chat box when that is open, but if the user taps on the window and no other taps have occurred, couldn't you assume that all keys should be forwarded to the x server using something like xdotool? I don't know if I have the technical knowledge necessary here, but I would like to try and help get this working. What if we take a look at how guacamole does things?

"Keyboard / input
I press [keyboard shortcut] but it’s handled by my browser or operating system instead of the remote desktop. Can you fix this?
Guacamole actually already attempts to handle all keyboard events. However, if your browser or operating system reserve some keyboard shortcuts for their own use. Keyboard events propagate from the operating system level, to the browser level, and finally into JavaScript where Guacamole resides. Guacamole can only take control of a keyboard event if each level above Guacamole allows it to do so.

Most browsers now provide a means of bookmarking a web application as a shortcut on the desktop or home screen such that it behaves more like a native application, lacks the normal URL bar, etc. In these cases, the browser will often allow the application to take control of additional keyboard shortcuts which would normally be reserved for the browser. If you are running into this problem, or simply want to use Guacamole as if it were a native application, this is definitely worth a try.

Does Guacamole support my keyboard layout?
Guacamole is actually independent of keyboard layout, and will send the true local identity of the key pressed. The behavior of each key within the remote desktop should identically match the local behavior of that key.

This holds true for most of Guacamole’s supported protocols, but things get more complex for RDP. Unlike Guacamole, RDP uses scancodes to represent each key, which are a numeric representation of that key’s location, not its identity. To translate the identity of the key into the scancode required by your RDP server, Guacamole must know the keyboard layout of your RDP server, and must have explicit support for that keyboard layout.

This does not mean that your RDP server must have the same keyboard layout as the client machine. It only means that Guacamole must have a keyboard layout definition for the layout used by the RDP server. If a user attempts to use a keyboard with a different layout, Guacamole will translate pressed keys as necessary to match the layout of the RDP server. If the user attempts to press keys which are not present in the keyboard layout of the RDP server, Guacamole will send those keys in the form of Unicode events.

If you are using an RDP server which does not support Unicode events (such as XRDP), locally-pressed keys must have matching keys within the RDP server’s keyboard layout, or they cannot be typed."
 https://guacamole.apache.org/faq/#:~:text=to%20the%20clipboard.-,Keyboard%20/%20input,the%20RDP%20server%E2%80%99s%20keyboard%20layout%2C%20or%20they%20cannot%20be%20typed.,-File%20transfer%20and

**Comment by alanmilinovic** [comment]
I would suggest not showing the keyboard at all, until working keyboard handling is developed.

**Comment by rosshaydenkelly** [comment]
> I would suggest not showing the keyboard at all (if possible), until working keyboard handling is developed.

@alanmilinovic Others can correct me here, but I could not find any way to disable the keyboard for mobile through config.
As a workaround for now, the simplest way I found to disable keyboard was to comment out "this._overlay.focus()" in the "onFucus()" method in video.vue and rebuild the client. Now when clicking the video on a mobile device, the keyboard no longer pops up for me. This works for my use case but is obviously a workaround and not a solution.

`onFocus() {
      // in order to capture key events, overlay must be focused
      if (this.focused && this.hosting && !this.locked) {
        //this._overlay.focus() 
      }
    }`

**Comment by alanmilinovic** [comment]
> > I would suggest not showing the keyboard at all (if possible), until working keyboard handling is developed.
> 
> @alanmilinovic Others can correct me here, but I could not find any way to disable the keyboard for mobile through config. As a workaround for now, the simplest way I found to disable keyboard was to comment out "this._overlay.focus()" in the "onFucus()" method in video.vue and rebuild the client. Now when clicking the video on a mobile device, the keyboard no longer pops up for me. This works for my use case but is obviously a workaround and not a solution.
> 
> `onFocus() { // in order to capture key events, overlay must be focused if (this.focused && this.hosting && !this.locked) { //this._overlay.focus() } }`

Nice finding, let's see what others will say.

**Comment by Xeddius** [comment]
I have a suggestion that can be implemented with relatively little effort.
We already have clipboard support and can type in the text box.
Why don't we just add a send button to the clipboard paste region which uses xdotool (or whatever is easiest) to ctrl+v to paste text? 
We can already highlight it by dragging so this should be enough to at the very least make it usable on mobile without intervention from a full system.

TL:DR add send button to clipboard text box to paste in n.eko so mobile has a working form of typing.

**Comment by m1k1o** [comment]
Implemented in https://github.com/m1k1o/neko/pull/497
